src/software/simulated_tests/validation_functions/robot_at_position_validation.cpp

```diff
@@ -0,0 +1,25 @@
+#include "software/logger/logger.h"
+#include "software/simulated_tests/validation_functions/robot_received_ball_validation.h"
+
+void robotAtPosition(RobotId robot_id, std::shared_ptr<World> world_ptr,
+                     const Point& destination, double close_to_destination_threshold,
+                     ValidationCoroutine::push_type& yield)
+{
+    auto robot_at_destination = [robot_id, destination, close_to_destination_threshold](
+                                    std::shared_ptr<World> world_ptr) {
+        std::optional<Robot> robotOptional =
+            world_ptr->friendlyTeam().getRobotById(robot_id);
+        if (!robotOptional.has_value())
+        {
+            LOG(FATAL) << "There is no robot with ID: " + std::to_string(robot_id);
```

Comment:
Is this preferred over an ASSERT here?

Reply:
I don't think we've use ASSERTs in our code outside of gtests before. LOG(FATAL) is more common in our codebase

Reply:
yeah agreed, but I guess this is more in the context of a validation function? so would `LOG(FATAL)` terminate the entire test suite rather than just this 1 test case?

not sure if it matters too much though